### PR TITLE
141: Added repo-wide linter

### LINKED ISSUE #141: Implement repo-wide linter
**Description**
In order to ensure our project's code is of good quality, implement a linter that checks all the different parts of the code. To do so, use [Mega-Linter](https://nvuillam.github.io/mega-linter/)

### PR DESCRIPTION
**Pull request by ZackHolmberg**
closes #141

### CODE REVIEW

**Comment by ZackHolmberg** [comment]
Generated logs:
[linters_logs.zip](https://github.com/ZackHolmberg/COMP4350-Project/files/6260921/linters_logs.zip)

File: .github/workflows/mega-linter.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+# Mega-Linter GitHub Action configuration file
+# More info at https://nvuillam.github.io/mega-linter
+name: Mega-Linter
+
+on:
+  # Trigger mega-linter at every push. Action will also be visible from Pull Requests to master
+  push: # Comment this line to trigger action only on pull-requests (not recommended if you don't pay for GH Actions)
+  pull_request:
+    branches: [master, main]
```
Comment (ZackHolmberg) [inline .github/workflows/mega-linter.yml]: remove main, add staging

**Comment by ZackHolmberg** [comment]
Screen Shot 2021-04-05 at 5 48 41 PM These auto-fix jobs are supposed to be running but I am not sure why they aren't. Regardless, this PR is only for adding the linter so it's good for review.

**Review by abhisachdev17 — CHANGES_REQUESTED** [review]
This looks good to me, before merging can you remove the flake8 linter from the python workflow and one concern: Is this going to be failing for every PR we create?

**Comment by abhisachdev17** [comment]
Also, do we need to run both flake8 and pylint for python?

**Review by madisonfines — APPROVED** [review]

**Review by abhisachdev17 — APPROVED** [review]
LGTM

**Pull request merged**